File: 020. Valid Parentheses/Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        
        stack<char> sk;
        for(int i = 0; i < s.size(); i++){
            if(s[i] == '(' || s[i] == '[' || s[i] == '{'){
                sk.push(s[i]);
            } else {
                if(sk.empty()) return false;
    
                char temp = sk.top();
                if(s[i] == ')' && temp != '(') return false;
                if(s[i] == ']' && temp != '[') return false;
                if(s[i] == '}' && temp != '{') return false;
                    
                sk.pop();
                
                
            }
           
        }
       return sk.empty();
    }
};

File: 020. Valid Parentheses/Valid Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> sk = new Stack<>();
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '(' || s.charAt(i) == '[' || s.charAt(i) == '{'){
                sk.push(s.charAt(i));
            } else {
                if(sk.empty()) return false;
    
                char temp = sk.peek();
                if(s.charAt(i) == ')' && temp != '(') return false;
                if(s.charAt(i) == ']' && temp != '[') return false;
                if(s.charAt(i) == '}' && temp != '{') return false;
                    
                sk.pop();
                
                
            }
           
        }
       return sk.empty();
    }
}
